msctl

```diff
@@ -1274,6 +1277,123 @@ serverConsole() {
   done
 }
 
+# ---------------------------------------------------------------------------
+# Retrieve the timestamp.
+#
+# @return The current date and time.
+# ---------------------------------------------------------------------------
+timestamp() {
+  date +"%Y-%m-%d_%H-%M-%S"
+}
+
+# ---------------------------------------------------------------------------
+# Stop the server monitor.
+#
+# @param 1 The world server to stop.
+# ---------------------------------------------------------------------------
+stopServerMonitor() {
+  local WORLD_DIR MONITOR_LOG MONITOR_PID MONITOR_LOCK_FILE ACQUIRED_LOCK
+  WORLD_DIR="$WORLDS_LOCATION/$1"
+  MONITOR_LOG="$WORLD_DIR/logs/mscs.monitor.log"
+  MONITOR_PID="$WORLD_DIR/monitor.pid"
+  MONITOR_LOCK_FILE="$WORLD_DIR/monitor.lock"
+  # Check if server monitor instance currently running.
+  (
+  flock -n 9
```

Comment:
`flock` should to be added to the `# Required Software` checks section and should use a variable alias

Reply:
There is a method in the script already called `createLockFile`. Would this work instead of using `flock`?

Reply:
Initially tried to use `createLockFile` instead of `flock` but it didn't work. If you are able to get it working without `flock` that would be great

Reply:
What didn't work when using `createLockFile`?

You are using `-n` which immediately exits with `1` and doesn't wait for a lock, so I think you could rewrite this to use `createLockFile` however, this function only creates a pid file

Reply:
`createLockFile` returns false when a lock exists, you might need to reverse the logic

Reply:
But it could return false if the lock file was not created...